File: base/src/main/java/com/qust/utils/CodeUtils.java
package com.qust.utils;

import androidx.annotation.NonNull;

public class CodeUtils{
	
	@NonNull
	public static String byteToHexString(@NonNull byte[] byteArray){
		StringBuilder hexString = new StringBuilder();
		for(byte b : byteArray){
			String hex = Integer.toHexString(0xFF & b);
			if(hex.length() == 1) hexString.append('0');
			hexString.append(hex);
		}
		return hexString.toString();
	}
	
	
	public interface Callback{
		void callback();
	}
}

File: app/src/main/java/com/qust/helper/utils/CodeUtils.kt
package com.qust.helper.utils

import java.util.regex.Pattern

object CodeUtils {

	fun byteToHexString(byteArray: ByteArray): String {
		val hexString = StringBuilder()
		for(b in byteArray) {
			val hex = Integer.toHexString(0xFF and b.toInt())
			if(hex.length == 1) hexString.append('0')
			hexString.append(hex)
		}
		return hexString.toString()
	}

	fun matcher(pattern: Pattern, string: String, index: Int = 1): String?{
		val matcher = pattern.matcher(string)
		return if(matcher.find()) matcher.group(1) else null
	}
}
